diff --git a/mlir/cmake/modules/AddMLIRPython.cmake b/mlir/cmake/modules/AddMLIRPython.cmake
index 012380603a4c..5a1d1c392281 100644
--- a/mlir/cmake/modules/AddMLIRPython.cmake
+++ b/mlir/cmake/modules/AddMLIRPython.cmake
@@ -328,6 +328,15 @@ function(declare_mlir_dialect_python_bindings)
       ADD_TO_PARENT "${_dialect_target}"
       SOURCES ${_sources}
     )
+
+    configure_file(
+      "${td_file}"
+      "${CMAKE_CURRENT_BINARY_DIR}/${ARG_TD_FILE}")
+    declare_mlir_python_sources("${_dialect_target}.python_bindings_td_file"
+      ROOT_DIR "${CMAKE_CURRENT_BINARY_DIR}"
+      ADD_TO_PARENT "${_dialect_target}"
+      SOURCES "${ARG_TD_FILE}"
+    )
   endif()
 endfunction()
 
@@ -399,6 +408,15 @@ function(declare_mlir_dialect_extension_python_bindings)
       ADD_TO_PARENT "${_extension_target}"
       SOURCES ${_sources}
     )
+
+    configure_file(
+      "${td_file}"
+      "${CMAKE_CURRENT_BINARY_DIR}/${ARG_TD_FILE}")
+    declare_mlir_python_sources("${ARG_DIALECT_NAME}.${ARG_EXTENSION_NAME}.python_bindings_td_file"
+      ROOT_DIR "${CMAKE_CURRENT_BINARY_DIR}"
+      ADD_TO_PARENT "${_extension_target}"
+      SOURCES "${ARG_TD_FILE}"
+    )
   endif()
 endfunction()
 
diff --git a/mlir/python/CMakeLists.txt b/mlir/python/CMakeLists.txt
index 5d2f233caa85..29f0ba7c2206 100644
--- a/mlir/python/CMakeLists.txt
+++ b/mlir/python/CMakeLists.txt
@@ -41,6 +41,11 @@ declare_mlir_python_sources(MLIRPythonCAPI.HeaderSources
   ROOT_DIR "${MLIR_SOURCE_DIR}/include"
   SOURCES_GLOB "mlir-c/*.h"
 )
+declare_mlir_python_sources(MLIRPythonCAPI.TableGenSources
+  ADD_TO_PARENT MLIRPythonCAPI.HeaderSources
+  ROOT_DIR "${MLIR_SOURCE_DIR}/include"
+  SOURCES_GLOB "mlir/*.td"
+)
 
 ################################################################################
 # Dialect bindings
