diff --git a/mlir/python/CMakeLists.txt b/mlir/python/CMakeLists.txt
index 39dd7b006c89..42621997e4b2 100644
--- a/mlir/python/CMakeLists.txt
+++ b/mlir/python/CMakeLists.txt
@@ -41,6 +41,11 @@ declare_mlir_python_sources(MLIRPythonCAPI.HeaderSources
   ROOT_DIR "${MLIR_SOURCE_DIR}/include"
   SOURCES_GLOB "mlir-c/*.h"
 )
+declare_mlir_python_sources(MLIRPythonCAPI.TableGenSources
+  ADD_TO_PARENT MLIRPythonCAPI.HeaderSources
+  ROOT_DIR "${MLIR_SOURCE_DIR}/include"
+  SOURCES_GLOB "mlir/*.td"
+)
 
 ################################################################################
 # Dialect bindings
