diff --git a/mlir/lib/ExecutionEngine/CMakeLists.txt b/mlir/lib/ExecutionEngine/CMakeLists.txt
index 39cc9c060db7..2bd0a6c55cdd 100644
--- a/mlir/lib/ExecutionEngine/CMakeLists.txt
+++ b/mlir/lib/ExecutionEngine/CMakeLists.txt
@@ -167,13 +167,13 @@ if(LLVM_ENABLE_PIC)
     LINK_LIBS PUBLIC
     ${LLVM_PTHREAD_LIB}
   )
-  set_property(TARGET mlir_async_runtime PROPERTY CXX_VISIBILITY_PRESET hidden)
+#  set_property(TARGET mlir_async_runtime PROPERTY CXX_VISIBILITY_PRESET hidden)
   target_compile_definitions(mlir_async_runtime PRIVATE mlir_async_runtime_EXPORTS)
   if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
     # Don't export symbols from link-time dependencies, these are internal
     # implementation details.
     # FIXME: Add a similar fix for Windows.
-    target_link_options(mlir_async_runtime PRIVATE "-Wl,-exclude-libs,ALL")
+#    target_link_options(mlir_async_runtime PRIVATE "-Wl,-exclude-libs,ALL")
   endif()
 
   if(MLIR_ENABLE_CUDA_RUNNER)
