diff --git a/cmake/Modules/CMakePolicy.cmake b/cmake/Modules/CMakePolicy.cmake
index cf986331707b..7f6c5a9113a3 100644
--- a/cmake/Modules/CMakePolicy.cmake
+++ b/cmake/Modules/CMakePolicy.cmake
@@ -42,6 +42,10 @@ if(POLICY CMP0156)
   endif()
 endif()
 
+if(POLICY CMP0091)
+  cmake_policy(SET CMP0091 NEW)
+endif()
+
 # CMP0182: Create shared library archives by default on AIX.
 # New in CMake 4.0: https://cmake.org/cmake/help/latest/policy/CMP0182.html
 if(POLICY CMP0182)
