diff --git a/mlir/lib/ExecutionEngine/CMakeLists.txt b/mlir/lib/ExecutionEngine/CMakeLists.txt
index fdeb4dacf927..aa17a07f4638 100644
--- a/mlir/lib/ExecutionEngine/CMakeLists.txt
+++ b/mlir/lib/ExecutionEngine/CMakeLists.txt
@@ -204,13 +204,13 @@ if(LLVM_ENABLE_PIC)
     LINK_LIBS PUBLIC
     ${LLVM_PTHREAD_LIB}
   )
-  set_property(TARGET mlir_async_runtime PROPERTY CXX_VISIBILITY_PRESET hidden)
+#  set_property(TARGET mlir_async_runtime PROPERTY CXX_VISIBILITY_PRESET hidden)
   target_compile_definitions(mlir_async_runtime PRIVATE mlir_async_runtime_EXPORTS)
   if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
     # Don't export symbols from link-time dependencies, these are internal
     # implementation details.
     # FIXME: Add a similar fix for Windows.
-    target_link_options(mlir_async_runtime PRIVATE "-Wl,-exclude-libs,ALL")
+#    target_link_options(mlir_async_runtime PRIVATE "-Wl,-exclude-libs,ALL")
   endif()
 
   add_mlir_library(mlir_arm_sme_abi_stubs
diff --git a/openmp/CMakeLists.txt b/openmp/CMakeLists.txt
index 88be8c75abfe..8559779ef882 100644
--- a/openmp/CMakeLists.txt
+++ b/openmp/CMakeLists.txt
@@ -170,5 +170,5 @@ else()
   add_subdirectory(docs)
 
   # Now that we have seen all testsuites, create the check-openmp target.
-  construct_check_openmp_target()
+#  construct_check_openmp_target()
 endif()
diff --git a/openmp/cmake/OpenMPTesting.cmake b/openmp/cmake/OpenMPTesting.cmake
index 60280b7ed489..df954deee0f3 100644
--- a/openmp/cmake/OpenMPTesting.cmake
+++ b/openmp/cmake/OpenMPTesting.cmake
@@ -1,5 +1,5 @@
 # Keep track if we have all dependencies.
-set(ENABLE_CHECK_TARGETS TRUE)
+#set(ENABLE_CHECK_TARGETS TRUE)
 
 # Function to find required dependencies for testing.
 function(find_standalone_test_dependencies)
