diff --git a/mlir/tools/mlir-tblgen/CMakeLists.txt b/mlir/tools/mlir-tblgen/CMakeLists.txt
index f2c5e4b3f87a..44e2e9b228e0 100644
--- a/mlir/tools/mlir-tblgen/CMakeLists.txt
+++ b/mlir/tools/mlir-tblgen/CMakeLists.txt
@@ -5,9 +5,7 @@ set(LLVM_LINK_COMPONENTS
   TableGen
 )
 
-add_tablegen(mlir-tblgen MLIR
-  DESTINATION "${MLIR_TOOLS_INSTALL_DIR}"
-  EXPORT MLIR
+set(MLIR_TBLGEN_LIBS
   AttrOrTypeDefGen.cpp
   AttrOrTypeFormatGen.cpp
   BytecodeDialectGen.cpp
@@ -31,6 +29,13 @@ add_tablegen(mlir-tblgen MLIR
   PassGen.cpp
   RewriterGen.cpp
   SPIRVUtilsGen.cpp
+)
+
+add_tablegen(mlir-tblgen MLIR
+  DESTINATION "${MLIR_TOOLS_INSTALL_DIR}"
+  EXPORT
+  MLIR
+  ${MLIR_TBLGEN_LIBS}
   )
 
 set_target_properties(mlir-tblgen PROPERTIES FOLDER "Tablegenning")
@@ -39,3 +44,33 @@ target_link_libraries(mlir-tblgen
   MLIRTblgenLib)
 
 mlir_check_all_link_libraries(mlir-tblgen)
+
+add_mlir_library(MLIRTableGenLib
+  ${MLIR_TBLGEN_LIBS}
+
+  EXCLUDE_FROM_LIBMLIR
+
+  DEPENDS
+  vt_gen
+
+  LINK_LIBS PUBLIC
+  MLIRTblgenLib
+)
+
+set(MLIR_TBLGEN_HEADERS
+  AttrOrTypeFormatGen.h
+  DialectGenUtilities.h
+  DocGenUtilities.h
+  FormatGen.h
+  OpClass.h
+  OpFormatGen.h
+  OpGenHelpers.h
+  )
+
+set_target_properties(MLIRTableGenLib
+  PROPERTIES PUBLIC_HEADER "${MLIR_TBLGEN_HEADERS}"
+  )
+install(TARGETS MLIRTableGenLib
+  PUBLIC_HEADER
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/mlir/Tools/mlir-tblgen
+  )
\ No newline at end of file
